src/WidgetMessaging.js

```diff
@@ -109,194 +109,297 @@ Example:
     action: "supported_api_versions",
 }
 
+
+OUTBOUND ACTIONS
+================
+
+In addition to listening for inbound requests, the API can be used to initiate
+actionss in the widget iframe, and request data from the widget instance.
+
+Outbound actions use the "widget_client" API key / name, which must be included
+on all requests.
+
+{
+    api: "widget_client",
+    action: "screenshot",
+    widgetId: $WIDGET_ID,
+    data: {}
+    // additional request fields
+}
+
+
+screenshot
+----------
+
+Request a screenshot from the widget (if supported).
+This can only be supported by widgets that have access to all of their DOM tree.
+For example, widgets that nest further levels of iframes can not support this.
+
+Request:
+ - No additional fields.
+Response:
+{
+    screenshot: {data...}
```

Comment:
What is `data`? a blob?

Reply:
Yup, a blob. I'll update the comment to reflect this.